jwst/outlier_detection/outlier_detection.py

```diff
@@ -131,81 +226,94 @@ def build_suffix(self, **pars):
+            The input data, DQ, and slice tuple output from ModelContainer.get_sections()
 
-        else:
-            # for non-dithered data, the resampled image is just the original image
-            drizzled_models = self.input_models
-            for i in range(len(self.input_models)):
-                drizzled_models[i].wht = build_driz_weight(
-                    self.input_models[i],
-                    weight_type=pars['weight_type'],
-                    good_bits=pars['good_bits'])
+        weight_thresholds : list
+            List of weight thresholds for each input image
 
-        # Initialize intermediate products used in the outlier detection
-        with datamodel_open(drizzled_models[0]) as dm0:
-            median_model = datamodels.ImageModel(dm0.data.shape)
-            median_model.update(dm0)
-            median_model.meta.wcs = dm0.meta.wcs
+        Returns
+        -------
+        np.ndarray
+            The median of the input data. Same dimensions as input.
+        """
+        sci, dq, _ = section
+
+        badmasks = []
+        for weight, weight_threshold in zip(dq, weight_thresholds):
+            badmask = np.less(weight, weight_threshold)
+            log.debug("Percentage of pixels with low weight: {}".format(
+                np.sum(badmask) / len(weight.flat) * 100))
+            badmasks.append(badmask)
+
+        # Fill resampled_sci array with nan's where mask values are True
+        for f1, f2 in zip(sci, badmasks):
+            for elem1, elem2 in zip(f1, f2):
+                elem1[elem2] = np.nan
+
+        # For a stack of images with "bad" data replaced with Nan
+        # use np.nanmedian to compute the median.
+        with warnings.catch_warnings():
+            warnings.filterwarnings(action="ignore",
+                                    message="All-NaN slice encountered",
+                                    category=RuntimeWarning)
+            median_image = np.nanmedian(sci, axis=0)
 
-        # Perform median combination on set of drizzled mosaics
-        median_model.data = self.create_median(drizzled_models)
-        if pars['save_intermediate_results']:
-            if self.outlierpars.get('asn_id', None) is None:
-                suffix_to_remove = self.resample_suffix
-            else:
-                suffix_to_remove = f"_{self.outlierpars['asn_id']}{self.resample_suffix}"
-            median_model_output_path = self.make_output_path(
-                basepath=median_model.meta.filename.replace(suffix_to_remove, '.fits'),
-                suffix='median')
-            median_model.save(median_model_output_path)
-            log.info(f"Saved model in {median_model_output_path}")
-        else:
-            # since we're not saving intermediate results if the drizzled models
-            # were written to disk, remove them
-            if not pars['in_memory']:
-                for fn in drizzled_models._models:
-                    _remove_file(fn)
```

Comment:
Similar to https://github.com/spacetelescope/jwst/pull/8473/files#r1600534514 I also saw leftover `_i2d` files in the working directory.